test/test_scheduler.py

```diff
@@ -1250,6 +1251,24 @@ def test_allocate_success(self):
         assert agent.gpus[1].resources["mem"].available == 1024.0
 
 
+class TestLogicalTask:
+    """Tests for :class:`katsdpcontroller.scheduler.LogicalTask`"""
+
+    def test_numa_nodes(self):
+        """Test the :attr:`~katsdpcontroller.scheduler.LogicalTask.numa_nodes` property."""
+        task = scheduler.LogicalTask("task")
+        assert task.numa_nodes == 0.0
+        task.numa_nodes = 0.5
+        assert task.numa_nodes == 0.5
+        with pytest.raises(ValueError):
+            task.numa_nodes = 1.2
+        with pytest.raises(ValueError):
+            task.numa_nodes = -0.1
+        with pytest.raises(ValueError):
+            task.numa_nodes = math.nan
+        assert task.numa_nodes == 0.5  # Must not have been altered by failures
```

Comment:
A small part of me thought "Why's he testing if the value is 0.5 twice?", but really that's the whole point of this addition/update.
* Need to check the initial update works correctly,
* Then check any erroneous updates/attempts to update don't affect it.

Reply:
Is there a change you're requesting here?

Reply:
No change requested, just thinking out aloud 😅